### PR TITLE
CMake Fix up prerelease version handling

### PR DESCRIPTION
**Pull request by yeswalrus**
#1611 Was authored before the introduction of examples.cmake. Upon testing, examples.cmake was broken and exposed some edge cases that weren't being handled.

This allows prerelease versions to be matched by find_package commands that do not specify a version, and adds lexographic comparison for determining compatible prerelease versions.

### CODE REVIEW

File: cmake/protobuf-config-version.cmake.in [hunk 1]
```diff
@@ -1,28 +1,35 @@
...
 endif()
 
 set(PACKAGE_VERSION_COMPATIBLE TRUE) #Assume true until shown otherwise
 
-# Handle prerelease versions
-set(PACKAGE_FIND_VERSION_PRERELEASE "${${PACKAGE_FIND_NAME}_FIND_VERSION_PRERELEASE}")
-if(NOT "@protobuf_VERSION_PRERELEASE@" STREQUAL "")
-  if(NOT ${PACKAGE_FIND_NAME}_FIND_VERSION_PRERELEASE)
+if(NOT PACKAGE_FIND_VERSION_MAJOR EQUAL "@protobuf_VERSION_MAJOR@")
+  set(PACKAGE_VERSION_COMPATIBLE FALSE)
```
Comment (anjaniied) [inline cmake/protobuf-config-version.cmake.in]: TRUE

Comment (yeswalrus) [inline cmake/protobuf-config-version.cmake.in]: No. If the major versions don't match, then the package is incompatible and this should be false.

**Comment by yeswalrus** [comment]
@xfxyjwf Ready for you

**Comment by yeswalrus** [comment]
@xfxyjwf ping

**Pull request merged**